### PR TITLE
support for multi-part file extensions like *.test.js

### PR DESCRIPTION
**Pull request by mrjones2014**
# Description

Updates code used to get icons from `nvim-web-devicons` to pass the filename as well as extension.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list relevant details about your configuration

Set up `nvim-web-devicons` with configuration:

```lua
require('nvim-web-devicons').setup({
  override = {
    ['js'] = {
      icon = '',
      color = '#efd81d',
      cterm_color = '220',
      name = 'Js',
    },
    ['test.js'] = {
      icon = 'ﭧ',
      color = '#ffca28',
      cterm_color = '220',
      name = 'TestJavascript',
    },
  },
})
```

Then create a `*.js` and a `*.test.js` file, and open `:Telescope find_file`, and make sure the icons appear as configured.

**Configuration**:
* Neovim version (nvim --version):
* Operating system and version:

# Checklist:

- [x] My code follows the style guidelines of this project (stylua)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (lua annotations)

### CODE REVIEW

File: lua/telescope/utils.lua [hunk 1]
```diff
@@ -446,7 +455,9 @@ utils.transform_devicons = load_once(function()
         return display
       end
 
-      local icon, icon_highlight = devicons.get_icon(utils.path_tail(filename), nil, { default = true })
+      local basename = utils.path_tail(filename)
+      -- the double :e:e gets multipart extensions, if one exists, like *.test.js, for example
```
Comment (Conni2461) [inline lua/telescope/utils.lua:459]: comment should be updated

File: lua/telescope/utils.lua [hunk 2]
```diff
@@ -476,7 +487,9 @@ utils.get_devicons = load_once(function()
         return ""
       end
 
-      local icon, icon_highlight = devicons.get_icon(utils.path_tail(filename), nil, { default = true })
+      local basename = utils.path_tail(filename)
+      -- the double :e:e gets multipart extensions, if one exists, like *.test.js, for example
```
Comment (Conni2461) [inline lua/telescope/utils.lua:491]: comment should be updated

File: lua/telescope/utils.lua (hunk 1 above)
Comment (mrjones2014) [reply lua/telescope/utils.lua]: Done.

File: lua/telescope/utils.lua (hunk 2 above)
Comment (mrjones2014) [reply lua/telescope/utils.lua]: done.

**Review by Conni2461 — APPROVED** [review]
LGTM thanks :)

@DasGandlaf if you want to improve on this, with what you said here https://github.com/nvim-telescope/telescope.nvim/pull/2235#discussion_r1035146006, feel free to open a new PR. Thanks :)

**Comment by numToStr** [comment]
FYI, this completely breaks the icons, at least for me. For example, `app.component.html` and `app.component.ts` no longer show html and typescript file icon to them respectively, which was not the case before this PR.

**Comment by Conni2461** [comment]
uff, does https://github.com/nvim-telescope/telescope.nvim/pull/2256 help?

**Comment by numToStr** [comment]
@Conni2461 Thanks a ton! It does fix the issue :)

**Pull request merged**